tests/api.rs

```diff
@@ -362,320 +445,306 @@ mod tests {
         assert_eq!(res.data.unwrap().meta.version_string, vector::get_version());
     }
 
-    #[tokio::test]
+    #[test]
     /// Tests that the heartbeat subscription returns a UTC payload every 1/2 second
-    async fn api_graphql_heartbeat() {
-        let server = start_server();
-        let client = new_subscription_client(server.addr()).await;
+    fn api_graphql_heartbeat() {
+        metrics_test("tests::api_graphql_heartbeat", async {
```

Comment:
In order for tests to be identified correctly across processes, the canonical test name should be provided here in the format of `mod::function`.

I had originally tried helper macros such as [function_name!](https://crates.io/crates/function_name) to make this play better with refactoring, but it added unnecessary boilerplate for what is a fairly simple copy/paste job.

Reply:
Yeah, think we can work with that.  Just have to be careful.  If you put `tests::some_random_name` in there it will happily just pass.

Reply:
Yeah, it's a shame there's no `nameof()` equivalent to keep these in sync. There are macros, but they wind up adding more LoC and still require developer buy-in, so I didn't feel they were adding much.

I'd have preferred an approach where `fork_test` knew the calling function and we could avoid this string name altogether, but I'm guessing any concept of 'caller' has been elided by the time this runs. I did poke around for a macro that might be able to fill that in based on usage, but turned up empty handed.

For now, I think this is fine, but it does rely on some manual checking.

Reply:
I wonder if we should structure `metrics_test` as macro? Like `#[metrics_test]`. I think that would resolve this issue as well.

Reply:
@jszwedko, love it! I think it'd need to be a proc-macro though, and be hosted in another crate. Open to moving forking to another lib for this purpose. Over to @lukesteensen for thoughts.

Reply:
It does seem like a proc macro with that helper crate could probably clean this up a bit, but honestly, I'm not sure it'd be worthwhile unless we plan to keep writing a lot more of these particular tests. The whole situation (global state, forking, etc) makes me a bit sad, so it'd be nice to limit the number of these kinds of tests we need to write and cover any additional ongoing testing needs with a type of test that's simpler and more direct.